OA-PASS/pass-docker
# Configure pass-docker with an LDAP-enabled mail server

## Purpose-built configuration of docker-mailserver for PASS.

- Accepts email for the `@jhu.edu` domain only.
- Any DN under `dc=People,dc=pass` with a `mail` attribute value ending in `@jhu.edu` is a valid recipient.  
    - i.e. any user defined in LDAP (see `users.ldif`) should be allowed to send mail, receive mail, and authenticate using IMAPS to check incoming mail.
- IMAP user names are the email address of the user
    - e.g. login with usernames like `staffWithGrants@jhu.edu` or `grant-admin-submitter@jhu.edu`
- Use port 587 for email submission [(RFC 4409)](https://tools.ietf.org/html/rfc4409)
    - no TLS, no SSL
    - by default published on port `11587` (per `.env` file)
- Use port 993 for IMAPS or TLS (authentication is disabled for unencrypted IMAP)
    - must use TLS or SSL
    - by default published on port `11993` (per `.env` file)
- Updates LDAP image containing additional user attributes

There are a couple of places (in the build, and at runtime) where parameters for Diffie-Hellman key exchange are generated.  Parameter sizes have been significantly reduced to minimize image build time and container startup time.  Therefore nothing in this mail server should be construed as occurring over secure channel(s).  Furthermore, the allowable SSL and TLS ciphers is generous, allowing a wider range of IMAP clients to connect using older and less secure algorithms.

## To Test

- Connect to the Mail Submission Port `11587` and attempt to send a message to an `@jhu.edu` recipient (expect success)
- Connect to the Mail Submission Port `11587` and attempt to send a message to a non `@jhu.edu` recipient (expect failure)
- Connect to the IMAPS port `11993` and attempt to login with a valid `@jhu.edu` email account (expect success)
- Connect to the IMAPS port `11993` and attempt to login with a invalid username or password (expect failure)

Instructions for each task are found [here](https://github.com/OA-PASS/docker-mailserver#testing-docker-mail-server).
